functions/go/starlark/go.mod

```diff
@@ -8,6 +8,8 @@ require (
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
 	k8s.io/api v0.21.0
 	k8s.io/apimachinery v0.21.0
-	sigs.k8s.io/kustomize/kyaml v0.10.21
+	// We use a unreleased version to pickup https://github.com/kubernetes-sigs/kustomize/pull/4023.
```

Comment:
@natasha41575 when are we planning to do the next release of kyaml ?
We would like to pin to the released versions.

Reply:
> when are we planning to do the next release of kyaml ?

I have asked @natasha41575 earlier this week. We don't have a clear answer yet.

> sigs.k8s.io/kustomize/kyaml v0.11.1-0.20210630191550-02d14d724aa6

The version used in this PR is less risky, since it is kyaml@v0.11.0 + the commit in kubernetes-sigs/kustomize#4023.